config/initializers/devise.rb

```diff
@@ -0,0 +1,313 @@
+# frozen_string_literal: true
+
+# Assuming you have not yet modified this file, each configuration option below
+# is set to its default value. Note that some are commented out while others
+# are not: uncommented lines are intended to protect your configuration from
+# breaking changes in upgrades (i.e., in the event that future versions of
+# Devise change the default values for those options).
+#
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
+  # random tokens. Changing this key will render invalid all existing
+  # confirmation, reset password and unlock tokens in the database.
+  # Devise will use the `secret_key_base` as its `secret_key`
+  # by default. You can change it below and use your own secret key.
+  # config.secret_key = '3b7592ae59968d71cc7eb4a4469c8529dbd072ab085bf3ec55f754924fd922ac5e55b9790e28769ac818cbcb6fab6535e5d10779889a79d4c966cd1c92b4914b'
```

Comment:
Where is the preferred ways to set environment variables?

1. In the `.env` file
2. In the `config/application.yml` file

Reply:
`application.yml` comes with the dependency [figjam](https://index.ruby-china.com/gems/figjam) so that should be used. There is ongoing effort to move to `.env` file in the Rails template but that is not done yet.

Reply:
Fixed in 1ef1d9f7f9618860ae2b904c4aba9e23e6eb0c72